### PR TITLE
[CI] Remove config that disables Bazel test result cache

### PR DESCRIPTION
**Pull request by mwtian**
## Why are these changes needed?
I believe there is no reason to disable Bazel test result cache. Removing these configs do not enable Bazel test result caching in CI yet, likely because of different environments in each CI run. But I think it is still useful to remove them first.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mwtian** [comment]
Windows failure seems unrelated.

File: .buildkite/Dockerfile [hunk 1]
```diff
@@ -48,7 +48,7 @@ RUN echo "ulimit -c 0" >> /root/.bashrc
 
 # Setup Bazel caches
 RUN (echo "build --remote_cache=${REMOTE_CACHE_URL}" >> /root/.bazelrc); \
-    (if [ "${BUILDKITE_PULL_REQUEST}" != "false" ]; then (echo "build --remote_upload_local_results=false" >> /root/.bazelrc); fi); \
```
Comment (simon-mo) [inline .buildkite/Dockerfile:51]: we need to keep the original flag as well. This is built so we are not uploading binary asserts to the cache.

Comment (mwtian) [reply .buildkite/Dockerfile:51]: Thanks for the info! Updated.

**Review by simon-mo — APPROVED** [review]
LGTM. To confirm this doesn't change how we are running tests in CI right? even though the builds are cached and the test results are not part of our current cache settting.

**Comment by mwtian** [comment]
Yes, so far no test result is cached actually, e.g. no cached annotation [here](https://buildkite.com/ray-project/ray-builders-pr/builds/15186#91888e8e-c837-436b-a746-5a0cb732b43a/217-258). Once I figure out how to actually cache test result, I will send in another PR and solicit more feedback from others.

**Comment by mwtian** [comment]
Actually the last comment is incorrect. I was not seeing cached test result because this is a PR branch. It seems master branch actually is able to upload test results to Bazel cache, and subsequent PRs can use the cached results. e.g. https://buildkite.com/ray-project/ray-builders-pr/builds/16501#d7d46eb7-4d3f-4e4d-a10b-2fbbefa4e308/216-331

**Comment by simon-mo** [comment]
ah nice!

**Pull request merged**